src/cryptsetup/cryptsetup-tpm2.c

```diff
@@ -64,7 +117,54 @@ int acquire_tpm2_key(
                 blob = loaded_blob;
         }
 
-        return tpm2_unseal(device, pcr_mask, pcr_bank, primary_alg, blob, blob_size, policy_hash, policy_hash_size, NULL, ret_decrypted_key, ret_decrypted_key_size);
+        if (!(flags & TPM2_FLAGS_USE_PIN)) {
+                return tpm2_unseal(
+                                device,
+                                pcr_mask,
+                                pcr_bank,
+                                primary_alg,
+                                blob,
+                                blob_size,
+                                policy_hash,
+                                policy_hash_size,
+                                NULL,
+                                ret_decrypted_key,
+                                ret_decrypted_key_size);
+        }
+
+        for (int i = 5;; i--) {
+                _cleanup_(erase_and_freep) char *pin_str = NULL;
+
+                if (i <= 0)
+                        return -EACCES;
+
+                r = get_pin(until, ask_password_flags, headless, &pin_str);
+                if (r == -EPERM)
```

Comment:
add comment what this means (but as mentioned, I think this should actually go away: do password length validation when setting PINs, not when checking PINS)